### PR TITLE
MGMT-8167: Move `assisted_service_api.py` from test_infra to be included in the service

### PR DESCRIPTION
**Pull request by eliorerz**
The `assisted_service_api.py` is basically part of `assisted_service_client` python library.
Now it is also needed to be as part as `assisted-events-scrap` project.
After this PR is merged, https://github.com/openshift/assisted-test-infra/blob/master/discovery-infra/test_infra/assisted_service_api.py and https://github.com/openshift-assisted/assisted-events-scrap/blob/master/ai_client/assisted_service_api.py will be removed and deprecated
## What environments does this code impact?

- [x] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [ ] None

## Assignees

/cc @osherdp 
/cc @michaellevy101

### CODE REVIEW

File: tools/client_package_initializer.py [hunk 1]
```diff
@@ -105,8 +113,23 @@ def _load_setup(self) -> None:
 
         self._setup_data.update(setup)
 
+    @classmethod
+    def _append_missing_requirements(cls, setup: Dict[str, List]):
+        # for assisted_service_api.py
+        if "kubernetes" not in "\n".join(setup[Setup.INSTALL_REQUIRES_KEY]):
+            setup[Setup.INSTALL_REQUIRES_KEY].append("kubernetes~=18.20.0")
```
Comment (osherdp) [inline tools/client_package_initializer.py:120]: please change to a broader range when listing dependencies in ``install_requires``

File: tools/client_package_initializer.py [hunk 2]
```diff
@@ -105,8 +113,23 @@ def _load_setup(self) -> None:
 
         self._setup_data.update(setup)
 
+    @classmethod
+    def _append_missing_requirements(cls, setup: Dict[str, List]):
+        # for assisted_service_api.py
```
Comment (osherdp) [inline tools/client_package_initializer.py:118]: replace with docstring (and some more explanation)

File: tools/client_package_initializer.py [hunk 3]
```diff
@@ -74,6 +77,10 @@ def __init__(self, project_path: str, url: str) -> None:
         with open(os.path.join(self._project_path, "MANIFEST.in"), "w") as f:
             f.write(f"include {self.DEFAULT_README_PATH}")
 
+    def _load_client_api(self) -> None:
```
Comment (osherdp) [inline tools/client_package_initializer.py:80]: the function name seems not indicative to what it does

File: tools/client_package_initializer.py [hunk 4]
```diff
@@ -148,11 +171,12 @@ def build(self) -> None:
 
 if __name__ == '__main__':
     parser = argparse.ArgumentParser(description="Swagger package builder")
+    parser.add_argument("client_extra_files_dir", help="External modules to add to the python api client", type=str)
```
Comment (osherdp) [inline tools/client_package_initializer.py:174]: any ``*extra*`` argument should come after the main one :) 
also, I recommend changing to named argument as its role is not clear when looking usage in ``hack/generate.sh``

File: tools/client_package_initializer.py [hunk 5]
```diff
@@ -105,8 +113,23 @@ def _load_setup(self) -> None:
...
+        if "kubernetes" not in "\n".join(setup[Setup.INSTALL_REQUIRES_KEY]):
+            setup[Setup.INSTALL_REQUIRES_KEY].append("kubernetes~=18.20.0")
+        if "waiting" not in "\n".join(setup[Setup.INSTALL_REQUIRES_KEY]):
+            setup[Setup.INSTALL_REQUIRES_KEY].append("waiting~=1.4.1")
+        if "requests" not in "\n".join(setup[Setup.INSTALL_REQUIRES_KEY]):
+            setup[Setup.INSTALL_REQUIRES_KEY].append("requests~=2.26.0")
+        if "urllib3" not in "\n".join(setup[Setup.INSTALL_REQUIRES_KEY]):
+            setup[Setup.INSTALL_REQUIRES_KEY].append("urllib3~=1.26.7")
+        if "retry" not in "\n".join(setup[Setup.INSTALL_REQUIRES_KEY]):
+            setup[Setup.INSTALL_REQUIRES_KEY].append("retry~=0.9.2")
```
Comment (osherdp) [inline tools/client_package_initializer.py:128]: there must be a shorter way to write this code

File: hack/generate.sh [hunk 6]
```diff
@@ -36,7 +36,9 @@ function generate_python_client() {
         -e SWAGGER_FILE=/swagger.yaml -e OUTPUT=/local/assisted-service-client/ \
         quay.io/ocpmetal/swagger-codegen-cli:2.4.15 /script.sh
      cd "${dest}"/assisted-service-client/ && python3 setup.py sdist --dist-dir "${dest}"
-     cd "${dest}"/assisted-service-client/ && python3 "${__root}"/tools/client_package_initializer.py "${dest}"/assisted-service-client/  https://github.com/openshift/assisted-service --build
+     cd "${dest}"/assisted-service-client/ && python3 "${__root}"/tools/client_package_initializer.py "${__root}"/tools/python_api_client_extra_files/ "${dest}"/assisted-service-client/  https://github.com/openshift/assisted-service --build
+     rm "${dest}"/assisted-service-client-1.0.0.tar.gz
+     cp "${dest}"/assisted-service-client/dist/assisted-service-client-*.tar.gz "${dest}"/
```
Comment (osherdp) [inline hack/generate.sh:41]:
```suggestion
     cp "${dest}"/assisted-service-client/dist/assisted-service-client-*.tar.gz "${dest}"
```

Comment (osherdp) [inline hack/generate.sh:41]: can you explain what happens here?

**Review by osherdp — CHANGES_REQUESTED** [review]

File: tools/client_package_initializer.py (hunk 5 above)
Comment (eliorerz) [reply tools/client_package_initializer.py:128]: Create another requirements file and update the current one

File: hack/generate.sh (hunk 6 above)
Comment (eliorerz) [reply hack/generate.sh:41]: 1. Updating the client and generating wheel and tar.gz artifacts
2. Remove the old artifact that was created so far
3. Copy the new source artifact to where the previous one was

File: tools/client_package_initializer.py (hunk 4 above)
Comment (eliorerz) [reply tools/client_package_initializer.py:174]: It's not extra argument. I can change the order but it's mandatory argument.

Comment (osherdp) [reply tools/client_package_initializer.py:174]: yeah, reorder
base always comes before extra :)

**Comment by nshidlin** [comment]
trying to understand what the followup PR in test-infra would look. where would the client be imported from?

**Comment by eliorerz** [comment]
> trying to understand what the followup PR in test-infra would look. where would the client be imported from?

```python
from assisted_service_client import assisted_service_api
```

**Comment by osherdp** [comment]
I still don't like all this ``package>=version`` in ``setup.py``, but I'll roll with that
/lgtm

**Comment by osherdp** [comment]
@eliorerz maybe you'll need to rebase

**Review by osherdp — APPROVED** [review]

**Comment by osherdp** [comment]
/hold
(just holding a bit while we still evaluating it)

File: tools/python_api_client_extra_files/assisted_service_api.py [hunk 7]
```diff
@@ -0,0 +1,533 @@
+# -*- coding: utf-8 -*-
+import base64
```
Comment (filanov) [inline tools/python_api_client_extra_files/assisted_service_api.py:2]: i don't think that this file belongs to assisted-service repo

**Comment by osherdp** [comment]
I agree, looking at scoping right now it does seem weird to have it here
sorry :) 
/lgtm cancel

**Review by osherdp — CHANGES_REQUESTED** [review]
maybe we can still use some stuff from this PR, but I agree the extra module should still exist in test-infra

**Comment by eliorerz** [comment]
After discussing it, we decided to leave the wrapper in assisted-test-infra repository but as separate package.
Opened new ticket - https://issues.redhat.com/browse/MGMT-8367

**Pull request closed without merging**